assets/js/admin-settings/rule.jsx

```diff
@@ -0,0 +1,145 @@
+/* global wp, wpLocalizedData */
+import {
+  Button,
+  SelectControl,
+  TextControl,
+  ToggleControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import React, { useEffect, useState } from 'react';
+import { useSelect } from '@wordpress/data';
+import { ruleCard } from './styles';
+
+
+const Rule = ({
+  field,
+  loading,
+  newRule,
+  onAdd,
+  onDelete,
+  onUpdate,
+  reorderRule,
+  ruleIndex,
+  saving,
+  setOriginIndex,
+  setTargetIndex,
+  taxonomy,
+  term_id,
+  value,
+}) => {
+  const {
+    fields,
+    taxonomies,
+  } = wpLocalizedData;
+
+  const [rule, setRule] = useState({
```

Comment:
Don't use state here - rely on the values passed in via props and update them using the update props. Let the state live in the parent component.